### PR TITLE
🌱 Add calico CRS

### PR DESCRIPTION
**Pull request by mcbenjemaa**
This PR adds:

* Calico CRS

### CODE REVIEW

File: docs/quickstart.md [hunk 1]
```diff
@@ -136,6 +143,55 @@ TODO(gfariasalves): Add instructions about installing a CNI or available flavour
 kubectl delete cluster ionos-quickstart
 ```
 
+## Cluster templates
+
+We provide various templates for creating clusters. Some of these templates provide you with a CNI already.
+
+For templates using `CNIs` you're required to create `ConfigMaps` to make `ClusterResourceSets` available.
+
+To enable `ClusterResourceSets` feature,
```
Comment (lubedacht) [inline docs/quickstart.md:152]: This is enabled by default.
https://cluster-api.sigs.k8s.io/tasks/experimental-features/cluster-resource-set.html?highlight=ClusterResource#experimental-feature-clusterresourceset-beta

File: docs/quickstart.md [hunk 2]
```diff
@@ -136,6 +143,55 @@ TODO(gfariasalves): Add instructions about installing a CNI or available flavour
...
+
+| Flavor         | Template File                          | CRS File                      |
+|----------------|----------------------------------------|-------------------------------|
+| calico         | templates/cluster-template-calico.yaml | templates/crs/cni/calico.yaml |
+| default        | templates/cluster-template.yaml        | -                             |
+
+
+#### Flavor with Calico CNI
+Before this cluster can be deployed, `calico` needs to be configured. As a first step we
+need to generate a manifest. Simply use our makefile:
```
Comment (lubedacht) [inline docs/quickstart.md:170]:
```suggestion
need to generate a manifest. Simply use our Makefile:
```

File: docs/quickstart.md [hunk 3]
```diff
@@ -136,6 +143,55 @@ TODO(gfariasalves): Add instructions about installing a CNI or available flavour
...
+make crs-calico
+
+```
+Now install the ConfigMap into your k8s:
+
+```
+kubectl create cm calico  --from-file=data=templates/crs/cni/calico.yaml
+```
+
+Now, you can create a cluster using the cilium flavor:
```
Comment (lubedacht) [inline docs/quickstart.md:182]:
```suggestion
Now, you can create a cluster using the calico flavor:
```

File: docs/quickstart.md (hunk 1 above)
Comment (mcbenjemaa) [reply docs/quickstart.md:152]: Good to know,

But this line won't have any affect whether to set it or not.

File: Makefile [hunk 4]
```diff
@@ -249,3 +249,15 @@ release-manifests: $(KUSTOMIZE) ## Create kustomized release manifest in $RELEAS
...
 	@mkdir -p $(RELEASE_DIR)
 	cp templates/cluster-template*.yaml $(RELEASE_DIR)/
+
+
+##@ CRS
+## --------------------------------------
+## CRS
+## --------------------------------------
+
+CALICO_VERSION ?= v3.26.3
```
Comment (avorima) [inline Makefile:259]: 3.28.0 is already out. can we choose a new version?

File: templates/cluster-template-calico.yaml [hunk 5]
```diff
@@ -0,0 +1,360 @@
...
+    name: "${CLUSTER_NAME}-control-plane"
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: "${CLUSTER_NAME}-credentials"
+type: Opaque
+stringData:
+  token: "${IONOS_TOKEN}"
+  apiURL: "${IONOS_API_URL:-https://api.ionos.com/cloudapi/v6}"
```
Comment (avorima) [inline templates/cluster-template-calico.yaml:28]: This PR has been open for a while so it missed some updates to the default template. Some fields were removed and others gained defaults. Also some things related to CCM integration were changed.
Please adjust this template accordingly

File: Makefile (hunk 4 above)
Comment (avorima) [reply Makefile:259]: ok, what about 3.27.x?

**Comment by mcbenjemaa** [comment]
Ready

File: Makefile (hunk 4 above)
Comment (avorima) [reply Makefile:259]: Yeah, the annotation value is not feasible for some things. Kamaji has the same problem. Maybe something like `kubectl create || kubectl replace` could work?

Comment (mcbenjemaa) [reply Makefile:259]: I don't see a point to discuss in which way we would create the configmap, 
as it depends on the setup.

The Kubectl create command is just for documentation, informing the user that it must be available in the cluster.
In production environments, the user will choose the solution to create the configmap.

**Review by avorima — APPROVED** [review]

**Review by lubedacht — APPROVED** [review]

**Pull request merged**